### PR TITLE
Allow search on rich text macros

### PR DESCRIPTION
**Pull request by PolRod**
The API response for rich text macros is different from those with plain text. This allows the app to search for rich text macros as well.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -401,11 +401,16 @@
...
         var comments = _.map(actions, function(action) {
-          return action.value[1].toLowerCase();
-        });
+          //Rich content macros' value is a string instead of an array. A string full of html.
+          if (typeof(action.value) == "string") {
+            return this.stripHTMLTags(action.value.toLowerCase());
+          } else {
+            return action.value[1].toLowerCase();
+          }
+        }.bind(this));
```
Comment (PolRod) [inline app.js:413]: `{this.("I did this with this this because binding this this allows this function to have access to this.stripHTMLTags.")}.bind(this)`

File: app.js [hunk 2]
```diff
@@ -442,6 +447,10 @@
 
       getEndDateQuery: function(queryType) {
         return new Date( this.$('.query.' + queryType + '.end-date').val().toLowerCase() );
+      }.bind(this),
+
+      stripHTMLTags: function(string) {
+        return string.replace(/<br>/, " ").replace(/<(?:.|\n)*?>/gm, '');
```
Comment (jpalmieri) [inline app.js:453]: so this replaces open `<br>` tags with spaces and then removes all other html tags?

File: app.js (hunk 1 above)
Comment (jpalmieri) [reply app.js:413]:
```javascript
this.("LGTM", function(yep) {
  return this.👍();
}.bind(this));
```

**Review by jpalmieri — CHANGES_REQUESTED** [review]
Won't this make the string search behave differently for rich text comments? Anything inside html tags will be removed (e.g., link addresses), so it won't be matched against the query, but only if the comment is rich text, right? Maybe I'm mistaken.

In other words, it seems like a plain text comment with the same content as a rich text comment could return more results; in the case that the query matched something that was inside an html tag, the result would return for the plain text comment but not for the rich text one, right?

However, it does seem weird to be matching stuff inside html tags anyway. But maybe we should figure out a solution that creates the same (or at least similar) behavior for both comment types, ideally without removing functionality. What's the problem that's being addressed by removing the html tags from the rich text comments?

File: app.js (hunk 2 above)
Comment (PolRod) [reply app.js:453]: Yeah, I thought it would make it more intuitive for searching things across lines.

**Comment by PolRod** [comment]
The idea is that an agent that is looking for a sentence in a rich text macro will see "This is a **cool** macro" on the editor and should not need to know to search for `"This is a <strong>cool</strong> macro"`. If we did not strip the tags, no results would come back for "this is a cool macro".

On the other hand, if there are tags on a plain text macro, they will show on the plain text editor, and the agent will expect to have to enter them when making a search.

In other words, I'm trying to make the results be the closest to what an agent would expect by looking at the macro.

**Comment by jpalmieri** [comment]
That totally makes sense. Thanks for clarifying! 🥇 💯

**Review by jpalmieri — APPROVED** [review]

**Comment by jpalmieri** [comment]
@PolRod I'm pretty sure you have merge permissions, so I'll let you merge whenever you want. But lmk if you can't. ✌️

**Comment by jpalmieri** [comment]
also, thanks for the contribution! 🙏

**Comment by jpalmieri** [comment]
forgot something. My bad @PolRod 

image
